sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -58,11 +58,11 @@ object JDBCRDD extends Logging {
     val dialect = JdbcDialects.get(url)
     val conn: Connection = JdbcUtils.createConnectionFactory(url, properties)()
     try {
-      val statement = conn.prepareStatement(s"SELECT * FROM $table WHERE 1=0")
+      val statement = conn.prepareStatement(dialect.getTableExistsQuery(table))
```

Comment:
I think that this might break schema inference for certain dialects because it's possible that the result schema of `getTableExistsQuery(table)` will be different than `select * from $table)`. Instead, we might need to define a new method in `JdbcDialect` to handle this schema-inference case. I think the goal of the `WHERE 1=0` was to coerece the remote JDBC database into telling us what the schema of the query's result set would be without actually returning any rows or executing the query (in theory we could use `LIMIT 1` but that could be expensive for, say, aggregate queries).